### PR TITLE
Implementation of change password, change profile, and delete user account function

### PR DESCRIPTION
**Pull request by frank2002**
## What does this pull request include?
**APIs (JSON format) for:**
Changing User password with the old password
Changing User profile when needed. This can be used to create a profile.
Users are able choose to delete their account.

**Simple forms (plain HTML):**
Changing User password with the old password
Changing User profile when needed. This can be used to create a profile.
Users are able choose to delete their account.

## Any outstanding issues in this pull request?
Not found yet

## Checklist

- [x ] Test locally
- [x ] Assign a reviewer
- [ x] ZenHub

### CODE REVIEW

File: Time_Banking/Time_Banking/templates/user_settings.html [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    <title>User Settings</title>
+</head>
+<body>
+    <h1>User Settings</h1>
+
+    <!-- Form for updating user information -->
+    <form id="userSettingsForm" enctype="multipart/form-data">
+        {% csrf_token %}  <!-- Include the CSRF token here -->
+
+        <label for="name">Name:</label>
```
Comment (austinhuang0131) [inline Time_Banking/Time_Banking/templates/user_settings.html:15]: If we have this already, do we still need the "create profile" page?

**Review by austinhuang0131 — APPROVED** [review]
We probably need to dump the entire tree at some point, `.gitignore`'ing the env might not be enough anymore

File: Time_Banking/Time_Banking/templates/user_settings.html (hunk 1 above)
Comment (frank2002) [reply Time_Banking/Time_Banking/templates/user_settings.html:15]: I think yes. We can use the settings as our profile creating page. But if there is too much forms to fill in the settings, we probably should have a seperate profile creating page. Anyway, the backend api can be reused. Only fronted end should be adjusted bit

**Comment by frank2002** [comment]
> We probably need to dump the entire tree at some point, `.gitignore`'ing the env might not be enough anymore
Yes, I think the `env` folder should be dump.
But this time I just untrack all the pyc files and accidentally caused that. Hopefully it will not cause trouble for anyone.

**Review by Nidhushan — COMMENTED** [review]
I tried change password and that seems to be working. Is delete account working? I couldnt find that.

**Comment by frank2002** [comment]
> I tried change password and that seems to be working. Is delete account working? I couldnt find that.

Yes they are working. I tested them locally. I should add the api entries in the description the next time. sorry for that.

you may test the delete function at:
http://127.0.0.1:8000/delete-account/

**Comment by Nidhushan** [comment]
> > I tried change password and that seems to be working. Is delete account working? I couldnt find that.
> 
> Yes they are working. I tested them locally. I should add the api entries in the description the next time. sorry for that.
> 
> you may test the delete function at: http://127.0.0.1:8000/delete-account/

Yes that works.

**Review by Nidhushan — APPROVED** [review]

**Pull request merged**